### PR TITLE
chore(storage): gRPC support customer encryption

### PR DESCRIPTION
**Pull request by noahdietz**
Enables use of customer provided encryption for Object media operations (Read, Rewrite, Write (including resumable), etc.).

Enables both `TestIntegration_Encryption` and `TestIntegration_KMS` for gRPC.

### CODE REVIEW

File: storage/grpc_client.go [hunk 1]
```diff
@@ -1511,7 +1511,10 @@ func (w *gRPCWriter) startResumableUpload() error {
 func (w *gRPCWriter) queryProgress() (int64, error) {
 	var persistedSize int64
 	err := run(w.ctx, func() error {
-		q, err := w.c.raw.QueryWriteStatus(w.ctx, &storagepb.QueryWriteStatusRequest{UploadId: w.upid})
+		q, err := w.c.raw.QueryWriteStatus(w.ctx, &storagepb.QueryWriteStatusRequest{
+			UploadId:                  w.upid,
+			CommonObjectRequestParams: toProtoCommonObjectRequestParams(w.encryptionKey),
```
Comment (tritone) [inline storage/grpc_client.go:1516]: Why does this RPC need encryptionKey?

**Review by tritone — COMMENTED** [review]
One small question, otherwise LGTM! Thanks Noah.

File: storage/grpc_client.go (hunk 1 above)
Comment (noahdietz) [reply storage/grpc_client.go:1516]: TBH IDK but it has the parameters. I can exclude it, but I figured I'd err on the side of caution. I can remove it though, wdyt?

Comment (noahdietz) [reply storage/grpc_client.go]: I've removed it, its just silly atm.

**Review by tritone — APPROVED** [review]

**Pull request merged**